File: src/main/java/com/bank/business/usecase/ShowCardBalance.java
package com.bank.business.usecase;

import com.bank.business.contract.cardReader.ReadCardData;
import com.bank.business.contract.display.DisplayError;
import com.bank.business.contract.display.DisplayMessage;
import com.bank.business.contract.display.GetCardPIN;
import com.bank.business.contract.bank.*;

public class ShowCardBalance {

    private GetCardBalance getCardBalance;
    private ReadCardData readCardData;
    private GetCardPIN getCardPIN;
    private ValidateCardPIN validateCardPIN;
    private DisplayMessage displayMessage;
    private DisplayError displayError;

    public ShowCardBalance(GetCardBalance getCardBalance,
                           ReadCardData readCardData,
                           GetCardPIN getCardPIN,
                           ValidateCardPIN validateCardPIN,
                           DisplayMessage displayMessage,
                           DisplayError displayError) {
        this.getCardBalance = getCardBalance;
        this.readCardData = readCardData;
        this.getCardPIN = getCardPIN;
        this.validateCardPIN = validateCardPIN;
        this.displayMessage = displayMessage;
        this.displayError = displayError;
    }

    public void invoke() {
        // todo: implement
        // displayMessage.invoke("Card balance: " + balance.amount() + " " + balance.currency());
    }
}

File: src/main/kotlin/com/bank/usecase/ShowCardBalance.kt
package com.bank.usecase

import com.bank.business.contract.bank.GetCardBalance
import com.bank.business.contract.bank.ValidateCardPIN
import com.bank.business.contract.cardReader.ReadCardData
import com.bank.business.contract.display.DisplayError
import com.bank.business.contract.display.DisplayMessage
import com.bank.business.contract.display.GetCardPIN

class ShowCardBalance(
    private val getCardBalance: GetCardBalance,
    private val readCardData: ReadCardData,
    private val getCardPIN: GetCardPIN,
    private val validateCardPIN: ValidateCardPIN,
    private val displayMessage: DisplayMessage,
    private val displayError: DisplayError
) {
    operator fun invoke() {
        // todo: implement
        // displayMessage("Card balance: " + balance.amount() + " " + balance.currency())
    }
}